tmt/libraries/beakerlib.py

```diff
@@ -330,6 +330,13 @@ def fetch(self) -> None:
                     self.parent.debug(f"Library {self} is copied into {directory}")
                     shutil.copytree(library_path, local_library_path, dirs_exist_ok=True)
 
+                    self.parent.verbose(
+                        'library',
+                        f"using remote repository: url='{self.url}', ref='{self.ref}'"
+                        f", path='{self.name}'",
```

Comment:
`'using remote repository', {'url': self.url, 'ref': self.ref, 'path': self.path}` might be better formatted, given it's a structured information.

Reply:
Yeah, that's better :), thanks.

Btw I just want to check this part: `'path': self.path`. I think we want to use the `'path': self.name` in this case. The `self.path` is empty.

```
/plans/example
summary: Basic smoke test
    discover
        how: fmf
        order: 50
        directory: /home/user/tmt-learning/03-beakertest
        using remote git library:
            url: https://github.com/beakerlib/tuned
            ref: master
            path: /basic
        using remote git library:
            url: https://github.com/beakerlib/openssl
            ref: devel
            path: /certgen
        summary: 1 test selected
            /tests/example
    report
        how: display
        order: 50
        summary: no results found
```

Reply:
Hm, maybe we want both `path` and `name`? It feels like an fmf ID, maybe it's already there somewhere or it could be constructed (https://github.com/teemtee/tmt/blob/main/tmt/base.py#L140) and its `to_minimal_spec()` logged, instead of composing it and checking for empty fields "manually".

Reply:
We can use the `self.identifier` which is just a `DependencySimple(str)` in the case of a "backward-compatible syntax" library name and the fields like `self.name`, `self.path`, etc. are parsed from it.

If I use a [full fmf identifier beaker library notation](https://tmt.readthedocs.io/en/stable/spec/tests.html#require), the type of `self.identifier` is a `DependencyFmfId` and it works as expected.

The output with your suggested changes is the following:

```
/plans/example
summary: Basic smoke test
    discover
        how: fmf
        order: 50
        directory: /home/user/tmt-learning/03-beakertest
        using remote git library: library(tuned/basic)
        using remote git library:
            url: https://github.com/beakerlib/openssl
            ref: devel
            name: /certgen
            type: library
        summary: 1 test selected
            /tests/example
```

I think we unfortunately must compose the fields manually, at least for "backward-compatible" library syntax.

Reply:
I agree, let's use the `name` as a name and `path` as a path :).

We can add a condition to print all the non-empty fields from `self.identifier` fmf id like the following, but I still think it would probably be better just to construct the dict "manually":

```
                    # By default, handle the backward-compatible library syntax
                    library_fields = {
                        'url': self.url,
                        'ref': self.ref,
                        'name': self.name}

                    if isinstance(self.identifier, DependencyFmfId):
                        library_fields = cast(dict[str, Optional[str]], self.identifier.to_minimal_dict())

                    self.parent.verbose('using remote git library', library_fields, 'green',
                                        level=3)

```

You choose.

Reply:
I can't shake off the feeling we're instantiating an fmf id :) It has the same fields, same meanings, how about this?

```python
fake_library_id = self.identifier if isinstance(self.identifier, DependencyFmfId) \
    else FmfId(url=self.url, ref=self.ref, path=self.path, name=self.name)

self.parent.verbose(
    'using remote git library',
    fake_library_id.to_minimal_dict(),
    'green',
    level=3)
```